start.go

```diff
@@ -23,17 +23,21 @@ type Config struct {
 
 	AllowedPrefixes []string // A list of metric prefixes to allow, with '.' as the separator
 	BlockedPrefixes []string // A list of metric prefixes to block, with '.' as the separator
+	AllowedLabels   []string // A list of metric labels to allow, with '.' as the separator
+	BlockedLabels   []string // A list of metric prefixes to block, with '.' as the separator
 	FilterDefault   bool     // Whether to allow metrics by default
 }
 
 // Metrics represents an instance of a metrics sink that can
 // be used to emit
 type Metrics struct {
 	Config
-	lastNumGC  uint32
-	sink       MetricSink
-	filter     *iradix.Tree
-	filterLock sync.RWMutex
+	lastNumGC     uint32
+	sink          MetricSink
+	filter        *iradix.Tree
+	allowedLabels map[string]int
+	blockedLabels map[string]int
+	filterLock    sync.RWMutex
```

Comment:
nitpick: I'd comment that all of `filter, allowedLabels, blockedLabels` are protected by `filterLock` as that's not implied by the name (no need to rename IMO just worth clarifying).

Reply:
DONE in next patch